### PR TITLE
pkg: convert `setup.py` config to `pyproject.toml`

### PR DESCRIPTION
**Pull request by shaypal5**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [metadata]
```
Comment (Borda) [inline pyproject.toml:1]: not sure if the metadata are used any more, would be better to convert to `project` tag
https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#basic-information

**Comment by shaypal5** [comment]
Ok, I'm lost in dynamic version loading hell, @Borda .

Please see if you can understand the correct way to dynamically provide the version under the `[project]` section of the `pyproject.toml` file, as it is mandatory once the `[project`] section is added.

See here for reference:
https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#basic-information

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
-description-file = "README.md"
+[project]
+name = "cachier"
+license = {file = "LICENSE"}
+readme = "README.rst"
+dynamic = ["version"]
+
+
+[tool.setuptools.dynamic]
+version = {attr = "cachier._version"}
```
Comment (Borda) [inline pyproject.toml:9]:
```suggestion
version = {attr = "cachier.__version__"}
```

File: pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,13 @@
...
-description-file = "README.md"
+[project]
+name = "cachier"
+license = {file = "LICENSE"}
+readme = "README.rst"
+dynamic = ["version"]
+
+
+[tool.setuptools.dynamic]
+version = {attr = "cachier.__version__"}
```
Comment (Borda) [inline pyproject.toml:9]:
```suggestion
version = {attr = "cachier._version.__version__"}
```

**Comment by Borda** [comment]
> Ok, I'm lost in dynamic version loading hell

will check it tomorrow

**Comment by shaypal5** [comment]
Actually this is the reference I wanted to send you, @Borda :
https://packaging.python.org/en/latest/guides/single-sourcing-package-version/#single-sourcing-the-version

**Comment by Borda** [comment]
@shaypal5 all fixed and in fact overwrote most of it :)

**Comment by shaypal5** [comment]
Awesome work, man! Thank you!❤️

**Pull request merged**